checker/src/main/java/org/checkerframework/checker/optional/RevisedOptionalChecker.java

```diff
@@ -0,0 +1,22 @@
+package org.checkerframework.checker.optional;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import org.checkerframework.checker.nonempty.NonEmptyChecker;
+import org.checkerframework.framework.source.AggregateChecker;
+import org.checkerframework.framework.source.SourceChecker;
+
+/** A version of the Optional Checker that runs the NonEmptyChecker as a subchecker. */
+// TODO: This is effectively the new Optional Checker and the name should reflect this fact.
```

Comment:
I think you should call this `OptionalChecker` and rename the current OptionalChecker.  (Maybe to OptionalWithoutNonEmptyChecker?  That's a gross name, but no one need see it, I hope.)

Reply:
I wonder if the `RevisedOptionalChecker` class is even needed?

The `NonEmptyChecker` running with the `-ArunAsOptionalChecker` flag enabled effectively behaves as the Optional Checker with higher precision wrt. container types.

I think it would make more sense to re-name the `NonEmptyChecker` to the `OptionalChecker`.

Reply:
What if a user only wants to check non-emptiness, not optionality?  For example, to ensure that get() and poll() are not called on empty collections.

Reply:
Is that something we want to support with the CF at this time (i.e., enable users to run the `NonEmptyChecker` by itself)?

I think there's still a lot of work to be done with the standalone Non-Empty Checker, running it solely as a checker for that use-case yields a large number of false positives that may not be acceptable by users. We can make this clear in the manual, however.

That said, I think having the `RevisedNonEmptyChecker` run the `NonEmptyChecker` as a subchecker might be  tricky; is there a way for a top-level checker to set the options of its child checkers (or the ones that belong to its aggregation, since the `RevisedOptionalChecker` extends `AggregateChecker`)? It would need to pass the `-ArunAsOptionalChecker` flag to its `NonEmptyChecker` instance.

Reply:
OK, let's skip that feature for now, and go with your proposal.  Thanks.